// @aws-sdk/client-cloudwatch-logs/dist-es/commands/CreateExportTaskCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateExportTaskRequest, CreateExportTaskResponse } from "../models/models_0";
import { deserializeAws_json1_1CreateExportTaskCommand, serializeAws_json1_1CreateExportTaskCommand, } from "../protocols/Aws_json1_1";
var CreateExportTaskCommand = (function (_super) {
    __extends(CreateExportTaskCommand, _super);
    function CreateExportTaskCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateExportTaskCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchLogsClient";
        var commandName = "CreateExportTaskCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateExportTaskRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateExportTaskResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateExportTaskCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateExportTaskCommand(input, context);
    };
    CreateExportTaskCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateExportTaskCommand(output, context);
    };
    return CreateExportTaskCommand;
}($Command));
export { CreateExportTaskCommand };
