// @aws-sdk/client-cloudwatch-logs/dist-es/commands/DescribeLogGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeLogGroupsRequest, DescribeLogGroupsResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeLogGroupsCommand, serializeAws_json1_1DescribeLogGroupsCommand, } from "../protocols/Aws_json1_1";
var DescribeLogGroupsCommand = (function (_super) {
    __extends(DescribeLogGroupsCommand, _super);
    function DescribeLogGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeLogGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchLogsClient";
        var commandName = "DescribeLogGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeLogGroupsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeLogGroupsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeLogGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeLogGroupsCommand(input, context);
    };
    DescribeLogGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeLogGroupsCommand(output, context);
    };
    return DescribeLogGroupsCommand;
}($Command));
export { DescribeLogGroupsCommand };
