// @aws-sdk/client-cloudwatch-logs/dist-es/commands/DescribeDestinationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeDestinationsRequest, DescribeDestinationsResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeDestinationsCommand, serializeAws_json1_1DescribeDestinationsCommand, } from "../protocols/Aws_json1_1";
var DescribeDestinationsCommand = (function (_super) {
    __extends(DescribeDestinationsCommand, _super);
    function DescribeDestinationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDestinationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchLogsClient";
        var commandName = "DescribeDestinationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDestinationsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeDestinationsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDestinationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeDestinationsCommand(input, context);
    };
    DescribeDestinationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeDestinationsCommand(output, context);
    };
    return DescribeDestinationsCommand;
}($Command));
export { DescribeDestinationsCommand };
