// @aws-sdk/client-cloudwatch-logs/dist-es/commands/GetLogEventsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetLogEventsRequest, GetLogEventsResponse } from "../models/models_0";
import { deserializeAws_json1_1GetLogEventsCommand, serializeAws_json1_1GetLogEventsCommand, } from "../protocols/Aws_json1_1";
var GetLogEventsCommand = (function (_super) {
    __extends(GetLogEventsCommand, _super);
    function GetLogEventsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetLogEventsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchLogsClient";
        var commandName = "GetLogEventsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetLogEventsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetLogEventsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetLogEventsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetLogEventsCommand(input, context);
    };
    GetLogEventsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetLogEventsCommand(output, context);
    };
    return GetLogEventsCommand;
}($Command));
export { GetLogEventsCommand };
