// @aws-sdk/client-cloudwatch-logs/dist-es/commands/ListTagsLogGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTagsLogGroupRequest, ListTagsLogGroupResponse } from "../models/models_0";
import { deserializeAws_json1_1ListTagsLogGroupCommand, serializeAws_json1_1ListTagsLogGroupCommand, } from "../protocols/Aws_json1_1";
var ListTagsLogGroupCommand = (function (_super) {
    __extends(ListTagsLogGroupCommand, _super);
    function ListTagsLogGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTagsLogGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CloudWatchLogsClient";
        var commandName = "ListTagsLogGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTagsLogGroupRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListTagsLogGroupResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTagsLogGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListTagsLogGroupCommand(input, context);
    };
    ListTagsLogGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListTagsLogGroupCommand(output, context);
    };
    return ListTagsLogGroupCommand;
}($Command));
export { ListTagsLogGroupCommand };
