// @aws-sdk/client-codepipeline/dist-es/commands/AcknowledgeJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AcknowledgeJobInput, AcknowledgeJobOutput } from "../models/models_0";
import { deserializeAws_json1_1AcknowledgeJobCommand, serializeAws_json1_1AcknowledgeJobCommand, } from "../protocols/Aws_json1_1";
var AcknowledgeJobCommand = (function (_super) {
    __extends(AcknowledgeJobCommand, _super);
    function AcknowledgeJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AcknowledgeJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "AcknowledgeJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AcknowledgeJobInput.filterSensitiveLog,
            outputFilterSensitiveLog: AcknowledgeJobOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AcknowledgeJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AcknowledgeJobCommand(input, context);
    };
    AcknowledgeJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AcknowledgeJobCommand(output, context);
    };
    return AcknowledgeJobCommand;
}($Command));
export { AcknowledgeJobCommand };
