// @aws-sdk/client-codepipeline/dist-es/commands/DeregisterWebhookWithThirdPartyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeregisterWebhookWithThirdPartyInput, DeregisterWebhookWithThirdPartyOutput } from "../models/models_0";
import { deserializeAws_json1_1DeregisterWebhookWithThirdPartyCommand, serializeAws_json1_1DeregisterWebhookWithThirdPartyCommand, } from "../protocols/Aws_json1_1";
var DeregisterWebhookWithThirdPartyCommand = (function (_super) {
    __extends(DeregisterWebhookWithThirdPartyCommand, _super);
    function DeregisterWebhookWithThirdPartyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeregisterWebhookWithThirdPartyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "DeregisterWebhookWithThirdPartyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeregisterWebhookWithThirdPartyInput.filterSensitiveLog,
            outputFilterSensitiveLog: DeregisterWebhookWithThirdPartyOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeregisterWebhookWithThirdPartyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeregisterWebhookWithThirdPartyCommand(input, context);
    };
    DeregisterWebhookWithThirdPartyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeregisterWebhookWithThirdPartyCommand(output, context);
    };
    return DeregisterWebhookWithThirdPartyCommand;
}($Command));
export { DeregisterWebhookWithThirdPartyCommand };
