// @aws-sdk/client-codepipeline/dist-es/commands/DisableStageTransitionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisableStageTransitionInput } from "../models/models_0";
import { deserializeAws_json1_1DisableStageTransitionCommand, serializeAws_json1_1DisableStageTransitionCommand, } from "../protocols/Aws_json1_1";
var DisableStageTransitionCommand = (function (_super) {
    __extends(DisableStageTransitionCommand, _super);
    function DisableStageTransitionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisableStageTransitionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "DisableStageTransitionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisableStageTransitionInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisableStageTransitionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DisableStageTransitionCommand(input, context);
    };
    DisableStageTransitionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DisableStageTransitionCommand(output, context);
    };
    return DisableStageTransitionCommand;
}($Command));
export { DisableStageTransitionCommand };
