// @aws-sdk/client-codepipeline/dist-es/commands/EnableStageTransitionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EnableStageTransitionInput } from "../models/models_0";
import { deserializeAws_json1_1EnableStageTransitionCommand, serializeAws_json1_1EnableStageTransitionCommand, } from "../protocols/Aws_json1_1";
var EnableStageTransitionCommand = (function (_super) {
    __extends(EnableStageTransitionCommand, _super);
    function EnableStageTransitionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    EnableStageTransitionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "EnableStageTransitionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: EnableStageTransitionInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    EnableStageTransitionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1EnableStageTransitionCommand(input, context);
    };
    EnableStageTransitionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1EnableStageTransitionCommand(output, context);
    };
    return EnableStageTransitionCommand;
}($Command));
export { EnableStageTransitionCommand };
