// @aws-sdk/client-codepipeline/dist-es/commands/GetPipelineExecutionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetPipelineExecutionInput, GetPipelineExecutionOutput } from "../models/models_0";
import { deserializeAws_json1_1GetPipelineExecutionCommand, serializeAws_json1_1GetPipelineExecutionCommand, } from "../protocols/Aws_json1_1";
var GetPipelineExecutionCommand = (function (_super) {
    __extends(GetPipelineExecutionCommand, _super);
    function GetPipelineExecutionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetPipelineExecutionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "GetPipelineExecutionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetPipelineExecutionInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetPipelineExecutionOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetPipelineExecutionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetPipelineExecutionCommand(input, context);
    };
    GetPipelineExecutionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetPipelineExecutionCommand(output, context);
    };
    return GetPipelineExecutionCommand;
}($Command));
export { GetPipelineExecutionCommand };
