// @aws-sdk/client-codepipeline/dist-es/commands/ListPipelinesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPipelinesInput, ListPipelinesOutput } from "../models/models_0";
import { deserializeAws_json1_1ListPipelinesCommand, serializeAws_json1_1ListPipelinesCommand, } from "../protocols/Aws_json1_1";
var ListPipelinesCommand = (function (_super) {
    __extends(ListPipelinesCommand, _super);
    function ListPipelinesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPipelinesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "ListPipelinesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPipelinesInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListPipelinesOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPipelinesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListPipelinesCommand(input, context);
    };
    ListPipelinesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListPipelinesCommand(output, context);
    };
    return ListPipelinesCommand;
}($Command));
export { ListPipelinesCommand };
