// @aws-sdk/client-codepipeline/dist-es/commands/ListActionExecutionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListActionExecutionsInput, ListActionExecutionsOutput } from "../models/models_0";
import { deserializeAws_json1_1ListActionExecutionsCommand, serializeAws_json1_1ListActionExecutionsCommand, } from "../protocols/Aws_json1_1";
var ListActionExecutionsCommand = (function (_super) {
    __extends(ListActionExecutionsCommand, _super);
    function ListActionExecutionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListActionExecutionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "ListActionExecutionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListActionExecutionsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListActionExecutionsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListActionExecutionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListActionExecutionsCommand(input, context);
    };
    ListActionExecutionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListActionExecutionsCommand(output, context);
    };
    return ListActionExecutionsCommand;
}($Command));
export { ListActionExecutionsCommand };
