// @aws-sdk/client-codepipeline/dist-es/commands/PollForJobsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PollForJobsInput, PollForJobsOutput } from "../models/models_0";
import { deserializeAws_json1_1PollForJobsCommand, serializeAws_json1_1PollForJobsCommand, } from "../protocols/Aws_json1_1";
var PollForJobsCommand = (function (_super) {
    __extends(PollForJobsCommand, _super);
    function PollForJobsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PollForJobsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "PollForJobsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PollForJobsInput.filterSensitiveLog,
            outputFilterSensitiveLog: PollForJobsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PollForJobsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1PollForJobsCommand(input, context);
    };
    PollForJobsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1PollForJobsCommand(output, context);
    };
    return PollForJobsCommand;
}($Command));
export { PollForJobsCommand };
