// @aws-sdk/client-codepipeline/dist-es/commands/PutJobSuccessResultCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutJobSuccessResultInput } from "../models/models_0";
import { deserializeAws_json1_1PutJobSuccessResultCommand, serializeAws_json1_1PutJobSuccessResultCommand, } from "../protocols/Aws_json1_1";
var PutJobSuccessResultCommand = (function (_super) {
    __extends(PutJobSuccessResultCommand, _super);
    function PutJobSuccessResultCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutJobSuccessResultCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "PutJobSuccessResultCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutJobSuccessResultInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutJobSuccessResultCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1PutJobSuccessResultCommand(input, context);
    };
    PutJobSuccessResultCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1PutJobSuccessResultCommand(output, context);
    };
    return PutJobSuccessResultCommand;
}($Command));
export { PutJobSuccessResultCommand };
