// @aws-sdk/client-codepipeline/dist-es/commands/PutJobFailureResultCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutJobFailureResultInput } from "../models/models_0";
import { deserializeAws_json1_1PutJobFailureResultCommand, serializeAws_json1_1PutJobFailureResultCommand, } from "../protocols/Aws_json1_1";
var PutJobFailureResultCommand = (function (_super) {
    __extends(PutJobFailureResultCommand, _super);
    function PutJobFailureResultCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutJobFailureResultCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "PutJobFailureResultCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutJobFailureResultInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutJobFailureResultCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1PutJobFailureResultCommand(input, context);
    };
    PutJobFailureResultCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1PutJobFailureResultCommand(output, context);
    };
    return PutJobFailureResultCommand;
}($Command));
export { PutJobFailureResultCommand };
