// @aws-sdk/client-codepipeline/dist-es/commands/UpdateActionTypeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateActionTypeInput } from "../models/models_0";
import { deserializeAws_json1_1UpdateActionTypeCommand, serializeAws_json1_1UpdateActionTypeCommand, } from "../protocols/Aws_json1_1";
var UpdateActionTypeCommand = (function (_super) {
    __extends(UpdateActionTypeCommand, _super);
    function UpdateActionTypeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateActionTypeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "UpdateActionTypeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateActionTypeInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateActionTypeCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateActionTypeCommand(input, context);
    };
    UpdateActionTypeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateActionTypeCommand(output, context);
    };
    return UpdateActionTypeCommand;
}($Command));
export { UpdateActionTypeCommand };
