// @aws-sdk/client-codepipeline/dist-es/commands/UpdatePipelineCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdatePipelineInput, UpdatePipelineOutput } from "../models/models_0";
import { deserializeAws_json1_1UpdatePipelineCommand, serializeAws_json1_1UpdatePipelineCommand, } from "../protocols/Aws_json1_1";
var UpdatePipelineCommand = (function (_super) {
    __extends(UpdatePipelineCommand, _super);
    function UpdatePipelineCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdatePipelineCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodePipelineClient";
        var commandName = "UpdatePipelineCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdatePipelineInput.filterSensitiveLog,
            outputFilterSensitiveLog: UpdatePipelineOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdatePipelineCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdatePipelineCommand(input, context);
    };
    UpdatePipelineCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdatePipelineCommand(output, context);
    };
    return UpdatePipelineCommand;
}($Command));
export { UpdatePipelineCommand };
