// @aws-sdk/client-codebuild/dist-es/commands/BatchGetProjectsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetProjectsInput, BatchGetProjectsOutput } from "../models/models_0";
import { deserializeAws_json1_1BatchGetProjectsCommand, serializeAws_json1_1BatchGetProjectsCommand, } from "../protocols/Aws_json1_1";
var BatchGetProjectsCommand = (function (_super) {
    __extends(BatchGetProjectsCommand, _super);
    function BatchGetProjectsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchGetProjectsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeBuildClient";
        var commandName = "BatchGetProjectsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchGetProjectsInput.filterSensitiveLog,
            outputFilterSensitiveLog: BatchGetProjectsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchGetProjectsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1BatchGetProjectsCommand(input, context);
    };
    BatchGetProjectsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1BatchGetProjectsCommand(output, context);
    };
    return BatchGetProjectsCommand;
}($Command));
export { BatchGetProjectsCommand };
