// @aws-sdk/client-codebuild/dist-es/commands/BatchGetReportsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetReportsInput, BatchGetReportsOutput } from "../models/models_0";
import { deserializeAws_json1_1BatchGetReportsCommand, serializeAws_json1_1BatchGetReportsCommand, } from "../protocols/Aws_json1_1";
var BatchGetReportsCommand = (function (_super) {
    __extends(BatchGetReportsCommand, _super);
    function BatchGetReportsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchGetReportsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeBuildClient";
        var commandName = "BatchGetReportsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchGetReportsInput.filterSensitiveLog,
            outputFilterSensitiveLog: BatchGetReportsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchGetReportsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1BatchGetReportsCommand(input, context);
    };
    BatchGetReportsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1BatchGetReportsCommand(output, context);
    };
    return BatchGetReportsCommand;
}($Command));
export { BatchGetReportsCommand };
