// @aws-sdk/client-codebuild/dist-es/commands/BatchGetBuildsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetBuildsInput, BatchGetBuildsOutput } from "../models/models_0";
import { deserializeAws_json1_1BatchGetBuildsCommand, serializeAws_json1_1BatchGetBuildsCommand, } from "../protocols/Aws_json1_1";
var BatchGetBuildsCommand = (function (_super) {
    __extends(BatchGetBuildsCommand, _super);
    function BatchGetBuildsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchGetBuildsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeBuildClient";
        var commandName = "BatchGetBuildsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchGetBuildsInput.filterSensitiveLog,
            outputFilterSensitiveLog: BatchGetBuildsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchGetBuildsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1BatchGetBuildsCommand(input, context);
    };
    BatchGetBuildsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1BatchGetBuildsCommand(output, context);
    };
    return BatchGetBuildsCommand;
}($Command));
export { BatchGetBuildsCommand };
