// @aws-sdk/client-codebuild/dist-es/commands/BatchGetReportGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetReportGroupsInput, BatchGetReportGroupsOutput } from "../models/models_0";
import { deserializeAws_json1_1BatchGetReportGroupsCommand, serializeAws_json1_1BatchGetReportGroupsCommand, } from "../protocols/Aws_json1_1";
var BatchGetReportGroupsCommand = (function (_super) {
    __extends(BatchGetReportGroupsCommand, _super);
    function BatchGetReportGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchGetReportGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeBuildClient";
        var commandName = "BatchGetReportGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchGetReportGroupsInput.filterSensitiveLog,
            outputFilterSensitiveLog: BatchGetReportGroupsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchGetReportGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1BatchGetReportGroupsCommand(input, context);
    };
    BatchGetReportGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1BatchGetReportGroupsCommand(output, context);
    };
    return BatchGetReportGroupsCommand;
}($Command));
export { BatchGetReportGroupsCommand };
