// @aws-sdk/client-codebuild/dist-es/commands/CreateReportGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateReportGroupInput, CreateReportGroupOutput } from "../models/models_0";
import { deserializeAws_json1_1CreateReportGroupCommand, serializeAws_json1_1CreateReportGroupCommand, } from "../protocols/Aws_json1_1";
var CreateReportGroupCommand = (function (_super) {
    __extends(CreateReportGroupCommand, _super);
    function CreateReportGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateReportGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeBuildClient";
        var commandName = "CreateReportGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateReportGroupInput.filterSensitiveLog,
            outputFilterSensitiveLog: CreateReportGroupOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateReportGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateReportGroupCommand(input, context);
    };
    CreateReportGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateReportGroupCommand(output, context);
    };
    return CreateReportGroupCommand;
}($Command));
export { CreateReportGroupCommand };
