// @aws-sdk/client-codebuild/dist-es/commands/ImportSourceCredentialsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ImportSourceCredentialsInput, ImportSourceCredentialsOutput } from "../models/models_0";
import { deserializeAws_json1_1ImportSourceCredentialsCommand, serializeAws_json1_1ImportSourceCredentialsCommand, } from "../protocols/Aws_json1_1";
var ImportSourceCredentialsCommand = (function (_super) {
    __extends(ImportSourceCredentialsCommand, _super);
    function ImportSourceCredentialsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ImportSourceCredentialsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeBuildClient";
        var commandName = "ImportSourceCredentialsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ImportSourceCredentialsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ImportSourceCredentialsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ImportSourceCredentialsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ImportSourceCredentialsCommand(input, context);
    };
    ImportSourceCredentialsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ImportSourceCredentialsCommand(output, context);
    };
    return ImportSourceCredentialsCommand;
}($Command));
export { ImportSourceCredentialsCommand };
