// @aws-sdk/client-codebuild/dist-es/commands/ListSharedReportGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListSharedReportGroupsInput, ListSharedReportGroupsOutput } from "../models/models_0";
import { deserializeAws_json1_1ListSharedReportGroupsCommand, serializeAws_json1_1ListSharedReportGroupsCommand, } from "../protocols/Aws_json1_1";
var ListSharedReportGroupsCommand = (function (_super) {
    __extends(ListSharedReportGroupsCommand, _super);
    function ListSharedReportGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListSharedReportGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeBuildClient";
        var commandName = "ListSharedReportGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSharedReportGroupsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListSharedReportGroupsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSharedReportGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListSharedReportGroupsCommand(input, context);
    };
    ListSharedReportGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListSharedReportGroupsCommand(output, context);
    };
    return ListSharedReportGroupsCommand;
}($Command));
export { ListSharedReportGroupsCommand };
