// @aws-sdk/client-codebuild/dist-es/commands/ListReportGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListReportGroupsInput, ListReportGroupsOutput } from "../models/models_0";
import { deserializeAws_json1_1ListReportGroupsCommand, serializeAws_json1_1ListReportGroupsCommand, } from "../protocols/Aws_json1_1";
var ListReportGroupsCommand = (function (_super) {
    __extends(ListReportGroupsCommand, _super);
    function ListReportGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListReportGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "CodeBuildClient";
        var commandName = "ListReportGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListReportGroupsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListReportGroupsOutput.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListReportGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListReportGroupsCommand(input, context);
    };
    ListReportGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListReportGroupsCommand(output, context);
    };
    return ListReportGroupsCommand;
}($Command));
export { ListReportGroupsCommand };
